packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+  which means that no clear pattern of object oriented programming such as `class`, `implements` (abstract class), and 'extends' (inheritance).  Used [util.inherits](https://nodejs.org/docs/latest/api/util.html#util_util_inherits_constructor_superconstructor) everywhere.
+- Not using [TypeScript](https://www.typescriptlang.org/)
+
+When I started to look around LoopBack.Next code base, I quickly got confused.  A couple of examples among many others:
+
+First, the LoopBack.Next source code does not look like JavaScript.  For example,
+
+- What are the [square brackets](http://es6-features.org/#ComputedPropertyNames) around object property name?
+```
+var obj = {
+  [key]: 'Ted Johnson',
+}
+```
+- I see how [the class](http://es6-features.org/#ClassDefinition) is defined.  My source code editor (such as [VS Code](https://code.visualstudio.com/) shows mouse-over pop-ups, which is super cool, but what's that [@inject](https://www.typescriptlang.org/docs/handbook/decorators.html)?
+```
+export class MyProvider implements Provider<Strategy> {
+  constructor(
+    @inject(bindingKey)
+    metadata: Metadata,
+  ) {}
+
+  value() : ValueOrPromise<Strategy> {}
+}
+```
+
+Secondly, there are several new technologies we use in LoopBack Next and each of them is significantly different from what I got used to.  Mentioned [ECMAScript 6](https://en.wikipedia.org/wiki/ECMAScript#6th_Edition_-_ECMAScript_2015) and [TypeScript](https://www.typescriptlang.org/) above; and [OpenAPI](https://www.openapis.org/) is another.  They are cool new technologies that make LoopBack Next really shining, but I felt being pulled out of my comfort zone many times a day.
+
+This blog will cover two major areas: Foundation and LoopBack Next programming.  I wanted to use half of the page space on the foundation technologies because "divide and conquer" is the right strategy here, i.e., start from the most basic stuff that has much less dependencies and build the next on top of it.  At the same time, we need to keep the clear goal in mind, i.e., LoopBack Next programming so that we can selectively study topics in the foundation technologies, move on to LoopBack Next space, then come back to the foundation when needed.
+
+In LoopBack Next topics, our focus is these three core concepts: controllers, components, and sequence.
+
+Alright, let's get the foundation techniques under the belt.
+
+
+
+# Foundation
+
+Here, I'm going to summarize a few key concepts and a list of tools I've found very useful.  With that, I got a big picture without getting bogged down to the details of each technology; and, at the same time, learned a way to go back the the details when needed.
+
+LoopBack Next is built on TypeScript and OpenAPI.  TypeScript is build on ECMAScript 6.  Quickly browsing those technologies from the ground up help solidify our steps. 
+
+### [ECMAScript 6](https://en.wikipedia.org/wiki/ECMAScript#6th_Edition_-_ECMAScript_2015)
```

Comment:
In fact, we are using ECMAScript 2017, which is an even newer version of the standard. (ES6 is an alias for ECMAScript 2015.)

The most notable language features we are using from ES2017 that are not in ES6:
 - `async`/`await`, a.k.a "async functions" (see https://tc39.github.io/ecmascript-asyncawait/)
 - trailing commas in function parameter lists and calls (see https://github.com/tc39/proposal-trailing-function-commas)

It would be great to mention at least "async functions" in the "Foundation" part, as we are using them pretty much everywhere in our codebase (and examples).